apps/shopify/src/productResolvers.js

```diff
@@ -30,11 +33,23 @@ export const fetchProductPreviews = async (skus, config) => {
     return [];
   }
 
-  const ids = skus.map(sku => `"${sku}"`).join(',');
+  const validIds = skus
+    .map(sku => {
+      try {
+        // If not valid base64 window.atob will throw
+        const unencodedId = atob(sku);
+        return { unencodedId, sku };
+      } catch (error) {
+        return null;
+      }
+    })
+    .filter(sku => sku && /^gid.*ProductVariant/.test(sku.unencodedId))
```

Comment:
What exactly is the regex testing? It seems like this functionality takes the skus, decodes them, tests the decoded value but then uses the encoded value finally.

Reply:
that's correct, the encoded value is what has to be passed in to the graphql query, but i need to decode them to determine which ones are validly formatted SKUs, then the rest (which are not) are put into the invalid SKUs array to be rendered as "missing products"